docs/tutorial/quick-start.md

```diff
@@ -0,0 +1,92 @@
+
+Once the personal access token is created, the charm can be deployed with:
+
+```shell
+juju deploy github-runner --constraints="cores=4 mem=16G" --config token=<TOKEN> --config path=<OWNER/REPO>
+```
+
+Replacing the `<TOKEN>` with the personal access token, and `<OWNER/REPO>` the GitHub account name and GitHub repository separated with `/`.
+
+The `--constraints` option for the `juju deploy` set the resource used for the juju machine hosting the charm application. This is used to accommodate different size of self-hosted runner.
+
+Once the charm is active status, visit the runner page for the GitHub repository (`https://github.com/{OWNER}/{REPO}/settings/actions/runners`) according to the instructions [here](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/using-self-hosted-runners-in-a-workflow#viewing-available-runners-for-a-repository). A single new runner should be available as it is the default number of self-hosted runners created.
+
+The charm would regularly goes into maintain status to managed the self-hosted runners. For example, replacing a self-hosted runner that was used up by a GitHub Actions job.
+
+### Ensure GitHub repository settings are secure
+
+For public repository, [arbitrary code execution with the self-hosted runners are possible](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners#self-hosted-runner-security). To combat this, the charm enforce a set of setting for the repository to ensure the code executed in the self-hosted runner are reviewed by someone trusted.
+
+Create a branch protection rule with the branch name pattern `**` and enable `Require signed commits` by following the instructions [here](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/managing-a-branch-protection-rule#creating-a-branch-protection-rule).
+
+### Run a simple workflow on the self-hosted runner
+
+Once the self-hosted runner is available on GitHub, it can be used to run GitHub Actions jobs similar to runners provided by GitHub. The only difference being the label specified in the `runs-on` of a job.
+
+The self-hosted runner managed by the charm would have the following labels: `self-hosted`, `linux`, and the application name
+
+In the above deployment, the application name was not specified hence the default value of `github-runner` was used. As such, `github-runner` would be a label for the self-hosted runner managed by the application instance.
+
+To test out the self-hosted runner, create the following file under the path `.github/workflows/runner_test.yaml` in the locally cloned git repository of the created GitHub repository with the following content:
```

Comment:
s/locally cloned git repository of the created GitHub repository with the/repository with the